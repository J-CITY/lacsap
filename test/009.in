


function fac(n: integer): integer;
var
  p, i: integer;
begin
  p:=1;
  for i:=2 to n do
    p:=p*i;
    
  fac:=p;
end;

procedure express(a, b, c : real);
var
  z : real;
begin
  z:=a + b + c;
  x:=sqr(z);
  y:=sqrt(z);
end ;

program HelloWorld;
var
    int : integer;
    r : real;
begin
    int := 1;
end.
