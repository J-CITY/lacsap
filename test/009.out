function 3 0 keyword 335
fac 3 9 identifier 4000
( 3 12 separator 8
n 3 13 identifier 4001
: 3 14 separator 5
integer 3 16 identifier 4002
) 3 23 separator 9
: 3 24 separator 5
integer 3 26 identifier 4002
; 3 33 separator 4
var 4 0 keyword 316
p 5 2 identifier 4003
, 5 3 separator 3
i 5 5 identifier 4004
: 5 6 separator 5
integer 5 8 identifier 4002
; 5 15 separator 4
begin 6 0 keyword 324
p 7 2 identifier 4003
:= 7 3 operation 23
1 7 5 literalInt 7000
; 7 6 separator 4
for 8 2 keyword 320
i 8 6 identifier 4004
:= 8 7 operation 23
2 8 9 literalInt 7001
to 8 11 keyword 314
n 8 14 identifier 4001
do 8 16 keyword 308
p 9 4 identifier 4003
:= 9 5 operation 23
p 9 7 identifier 4003
* 9 8 operation 0
i 9 9 identifier 4004
; 9 10 separator 4
fac 11 2 identifier 4000
:= 11 5 operation 23
p 11 7 identifier 4003
; 11 8 separator 4
end 12 0 keyword 315
; 12 3 separator 4
procedure 14 0 keyword 336
express 14 10 identifier 4005
( 14 17 separator 8
a 14 18 identifier 4006
, 14 19 separator 3
b 14 21 identifier 4007
, 14 22 separator 3
c 14 24 identifier 4008
: 14 26 separator 5
real 14 28 identifier 4009
) 14 32 separator 9
; 14 33 separator 4
var 15 0 keyword 316
z 16 2 identifier 4010
: 16 4 separator 5
real 16 6 identifier 4009
; 16 10 separator 4
begin 17 0 keyword 324
z 18 2 identifier 4010
:= 18 3 operation 23
a 18 5 identifier 4006
+ 18 7 operation 19
b 18 9 identifier 4007
+ 18 11 operation 19
c 18 13 identifier 4008
; 18 14 separator 4
x 19 2 identifier 4011
:= 19 3 operation 23
sqr 19 5 identifier 4012
( 19 8 separator 8
z 19 9 identifier 4010
) 19 10 separator 9
; 19 11 separator 4
y 20 2 identifier 4013
:= 20 3 operation 23
sqrt 20 5 identifier 4014
( 20 9 separator 8
z 20 10 identifier 4010
) 20 11 separator 9
; 20 12 separator 4
end 21 0 keyword 315
; 21 4 separator 4
program 23 0 keyword 334
HelloWorld 23 8 identifier 4015
; 23 18 separator 4
var 24 0 keyword 316
int 25 4 identifier 4016
: 25 8 separator 5
integer 25 10 identifier 4002
; 25 17 separator 4
r 26 4 identifier 4017
: 26 6 separator 5
real 26 8 identifier 4009
; 26 12 separator 4
begin 27 0 keyword 324
int 28 4 identifier 4016
:= 28 8 operation 23
1 28 11 literalInt 7002
; 28 12 separator 4
end 29 0 keyword 315
. 29 3 separator 6
